api/app/tests/fba/test_fba_error_random_sample.py

```diff
@@ -0,0 +1,239 @@
+"""
+Unit tests for fire behavour calculator.
+"""
+from datetime import datetime, timezone as dt_tz
+import random
+from typing import Final
+import logging
+from pytest_bdd import scenario, given, then
+from app import configure_logging
+from app.utils.time import get_hour_20_from_date
+from app.utils.fba_calculator import calculate_fire_behaviour_advisory, FBACalculatorWeatherStation
+from app.utils.redapp import FBPCalculateStatisticsCOM
+from app.utils.cffdrs import initial_spread_index, bui_calc
+import pytest
+
+
+configure_logging()
+
+logger = logging.getLogger(__name__)
+
+
+def _str2float(value: str):
+    if value == 'None':
+        return None
+    return float(value)
+
+
+def _random_date():
+    start = datetime.fromisoformat('2021-01-01')
+    end = datetime.fromisoformat('2021-12-31')
+    return datetime.fromtimestamp(random.uniform(start.timestamp(), end.timestamp()), tz=dt_tz.utc)
+
+
+def relative_error(actual: float, expected: float, precision: int = 2):
+    """ calculate the relative error between two values - default to precision of 2"""
+    actual = round(actual, precision)
+    expected = round(expected, precision)
+    if actual == expected:
+        return 0
+    if expected == 0:
+        # Can't divide by 0! Taking this as a 100% difference
+        return 1
+    return abs((actual-expected)/expected)
+
+
+acceptable_margin_of_error: Final = 0.01
+
+
+def check_metric(metric: str,
+                 fuel_type: str,
+                 python_value: float,
+                 comparison_value: float,
+                 metric_error_margin: float, note: str = None):
+    """ Check relative error of a metric """
+    # logging with %s became unreadable:
+    # pylint: disable=logging-fstring-interpolation
+    if comparison_value < 0:
+        logger.warning('Skipping %s! (%s) - note: %s', metric, comparison_value, note)
+    else:
+        assert python_value >= 0
+        error = relative_error(python_value, comparison_value)
+        absolute_error = abs(python_value - comparison_value)
+        # logger.info(
+        #     f'{fuel_type}: Python {python_value}, {metric} {comparison_value} ; error: {error}')
```

Comment:
did you intend to keep this commented out?

Reply:
Will remove.